Many-body-guess.py

```diff
@@ -0,0 +1,133 @@
+       ("H", -0.093821, 2.209720, -0.643619),
+       ("O", 1.517569, -0.667424, -0.080674),
+       ("H", 1.989645, -1.098799, 0.612047),
+       ("H", 0.668397, -1.091798, -0.139744),
+       ("O", -1.350388, -0.964879, -0.092208),
+       ("H", -1.908991, -1.211298, 0.626207),
+       ("H", -1.263787, -0.018107, -0.055536)]
+
+bs = "aug-cc-pvdz"
+H2OA = density(mol[0:3], bs)
+H2OB = density(mol[3:6], bs)
+H2OC = density(mol[6:9], bs)
+H4O2AB = density(mol[0:6], bs)
+H4O2AC = density((mol[0:3]+mol[6:9]), bs)
+H4O2BC = density(mol[3:9], bs)
+#density_list = [H2OA, H2OB, H2OC, H4O2AB, H4O2AC, H4O2BC]
+
+N,N = H4O2AB.shape
+M,M = H2OA.shape
+A = np.zeros((M*3,M*3))
+
+#Monomer_density = block_diag(H2OA, H2OB, H2OC)
+A[0:M, 0:M] = H4O2AB[0:M, 0:M]+H4O2AC[0:M, 0:M]-H2OA
+A[M:N, M:N] = H4O2AB[M:N, M:N]+H4O2BC[0:M, 0:M]-H2OB
+A[N:M*3, N:M*3] = H4O2AC[M:N, M:N]+H4O2BC[M:N, M:N]-H2OC
+A[0:M, M:N] = H4O2AB[0:M, M:N]
+A[M:N, 0:M] = H4O2AB[M:N, 0:M]
+A[0:M, N:M*3] = H4O2AC[0:M, M:N]
+A[N:M*3, 0:M] = H4O2AC[M:N, 0:M]
+A[M:N, N:M*3] = H4O2BC[0:M, M:N]
+A[N:M*3, M:N] = H4O2BC[M:N, 0:M]
+
+rho = A
+nitr = run_scf(mol, bs, rho)
+print(nitr)
+
+
+from wrap_pyscf import *
+import numpy as np
+
+mol = [("O", 0.216486, -1.989005, 0.035297),
+       ("H", -0.603911, -1.505274, -0.033552),
+       ("H", 0.138672, -2.535875, 0.799566),
+       ("H", 1.505276, -0.603920, 0.033505),
+       ("O", 1.989004, 0.216483, -0.035296),
+       ("H", 2.535922, 0.138690, -0.799534),
+       ("O", -1.989001, -0.216487, -0.035297),
+       ("H", -2.535947, -0.138688, -0.799513),
+       ("H", -1.505273, 0.603920, 0.033493),
+       ("O", -0.216485, 1.989007, 0.035294),
+       ("H", -0.138694, 2.535879, 0.799565),
+       ("H", 0.603919, 1.505277, -0.033513)]
+
+bs = "aug-cc-pvdz"
+H2OA = density(mol[0:3], bs)
+H2OB = density(mol[3:6], bs)
+H2OC = density(mol[6:9], bs)
+H2OD = density(mol[9:12], bs)
+H4O2AB = density(mol[0:6], bs)
+H4O2AC = density((mol[0:3]+mol[6:9]), bs)
+H4O2AD = density((mol[0:3]+mol[9:12]), bs)
+H4O2BC = density(mol[3:9], bs)
+H4O2BD = density((mol[3:6]+mol[9:12]), bs)
+H4O2CD = density(mol[6:12], bs)
+
+N,N = H4O2AB.shape
+M,M = H2OA.shape
+A = np.zeros((N*2, N*2))
+
+A[0:M, 0:M] = H4O2AB[0:M, 0:M]+H4O2AC[0:M, 0:M]+H4O2AD[0:M, 0:M]-2*H2OA
+A[M:N, M:N] = H4O2AB[M:N, M:N]+H4O2BC[0:M, 0:M]+H4O2BD[0:M, 0:M]-2*H2OB
+A[N:M*3, N:M*3] = H4O2AC[M:N, M:N]+H4O2BC[M:N, M:N]+H4O2CD[0:M, 0:M]-2*H2OC
+A[M*3:N*2, M*3:N*2] = H4O2AD[M:N, M:N]+H4O2BD[M:N, M:N]+H4O2CD[M:N, M:N]-2*H2OD
+A[0:M, M:N] = H4O2AB[0:M, M:N]
+A[M:N, 0:M] = H4O2AB[M:N, 0:M]
+A[0:M, N:M*3] = H4O2AC[0:M, M:N]
+A[N:M*3, 0:M] = H4O2AC[M:N, 0:M]
+A[M:N, N:M*3] = H4O2BC[0:M, M:N]
+A[N:M*3, M:N] = H4O2BC[M:N, 0:M]
+A[M*3:N*2, 0:M] = H4O2AD[M:N, 0:M]
+A[0:M, M*3:N*2] = H4O2AD[0:M, M:N]
+A[M*3:N*2, M:N] = H4O2BD[M:N, 0:M]
+A[M:N, M*3:N*2] = H4O2BD[0:M, M:N]
+A[M*3:N*2, N:M*3] = H4O2CD[M:N, 0:M]
+A[M*3:N*2, N:M*3] = H4O2CD[0:M, M:N]
+
+rho = A
+nitr = run_scf(mol, bs, rho)
+print(nitr)
```

Comment:
Unfortunately I got 9 cycles with this script, which is the same as the atomic guess. There could be an error in the script, but I haven't found one. 

If considering dimer interactions improved the trimer guess, would trimer interactions improve the guess for the tetramer?

Reply:
Reasonable that the trimers would help the tetramer guess. You could kind of stack the process. Use monomers to guess dimers, dimers for trimers, and so on.

Reply:
If we have to go to trimers to get a good guess this idea is dead in the water (no pun intended). The MBE is pretty converged by the three-body term for most systems (water's significant polarization makes it an exception; for water it converges with four-body terms). In other words, if we have to do trimers the work required to get the guess is the same amount of work required to compute the energy of the full system (for most systems)...

Reply:
Next I'd try writing the more general two-body algorithm and applying it to the water pentamer and hexamer.